src/PregWrapper.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer;
+
+class PregWrapper
+{
+    public static function replace($pattern, $replacement, $subject)
+    {
+        // TODO: check if "u" modifier is in the pattern and throw exception if so
+
+        $newSubject = preg_replace($pattern.'u', $replacement, $subject);
+
+        if (null === $newSubject && PREG_BAD_UTF8_ERROR === preg_last_error()) {
+            $newSubject = preg_replace($pattern, $replacement, $subject);
+        }
+
+        return $newSubject;
```

Comment:
I would love if we would check `null === $newSubject` for error, if we are already wrapping internal functions into OOP, let us use expections over ANSI-C style errors